Fix __cxa_find_matching_catch's memory leak

aheejin opened the pull request:
`__cxa_find_matching_catch` allocated a buffer of size 4 using `malloc` only
when it was first called and reused it through the entire program run. And this
malloc'ed memory wasn't freed. Because this is only allocated once, I changed it
to a static allocation using `makeStaticAlloc`.                                  
                                                                                 
Fixes #8919.

kripken commented:
Hmm that CI error on `test_clear_error_on_massive_static_data` is worrying. There should be no change to static memory from this PR, but I think that error means that there is.

I think the issue is that we include that JS file based on `DISABLE_EXCEPTION_THROWING` and not `_CATCHING`. I don't remember all the details for the difference (some notes are in settings.js). But I think we do need to ifdef this on `DISABLE_EXCEPTION_CATCHING` (which is on by default, unlike the other one).

aheejin commented:
Not sure why 

> Hmm that CI error on `test_clear_error_on_massive_static_data` is worrying. There should be no change to static memory from this PR, but I think that error means that there is.
> 
> I think the issue is that we include that JS file based on `DISABLE_EXCEPTION_THROWING` and not `_CATCHING`. I don't remember all the details for the difference (some notes are in settings.js). But I think we do need to ifdef this on `DISABLE_EXCEPTION_CATCHING` (which is on by default, unlike the other one).

`test_clear_error_on_massive_static_data` works on my local machine, so it's weird. I addressed feedback and merged incoming just now, so I'll wait the CI to complete and see that happens again.
